src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -117,5 +147,2211 @@ public void TexlDateTableFunctions_Negative(string expression)
+            TestSimpleBindingSuccess(
+                "Trunc(T)",
+                TestUtils.DT("*[A:n]"),
+                symbol);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsTruncOneParam_ConsistentOneColumnTableResult()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[A:n]")));
+            TestSimpleBindingSuccess(
+                "Trunc(T)",
+                TestUtils.DT("*[A:n]"),
+                symbol,
+                Features.ConsistentOneColumnTableResult);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsTruncTwoParams()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[digits:n]")));
+            symbol.AddVariable("X", new TableType(TestUtils.DT("*[Nnnuuummm:n]")));
+
+            TestSimpleBindingSuccess(
+                "Trunc(1234.567, 4)",
+                DType.Number);
+
+            TestSimpleBindingSuccess(
+                "Trunc(1234.567, T)",
+                TestUtils.DT("*[Result:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "Trunc(X, T)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "Trunc(X, 4)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsRound()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[digits:n]")));
+            symbol.AddVariable("X", new TableType(TestUtils.DT("*[Nnnuuummm:n]")));
+
+            TestSimpleBindingSuccess(
+                "Round(1234.567, 4)",
+                DType.Number);
+
+            TestSimpleBindingSuccess(
+                "Round(1234.567, T)",
+                TestUtils.DT("*[Result:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "Round(X, T)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "Round(X, 4)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+        }
+
+        [Theory]
+        [InlineData("Round(1234.567, T)", "*[Value:n]")]
+        [InlineData("Round(4, X)", "*[Value:n]")]
+        [InlineData("Round(X, 4)", "*[Nnnuuummm:n]")]
+        public void TexlFunctionTypeSemanticsRound_ConsistentOneColumnTableResult(string expression, string expectedType)
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[digits:n]")));
+            symbol.AddVariable("X", new TableType(TestUtils.DT("*[Nnnuuummm:n]")));
+
+            TestSimpleBindingSuccess(
+                expression,
+                TestUtils.DT(expectedType),
```

Comment:
>expectedType [](http://example.com/codeflow?start=29&length=12)

It should always be `*[Value:n]` if we're using the ConsistentOneColumnTableResult flag

Reply:
So, is it okay to remove the third test, Round(X,4)? I just had it to show difference of having table on arg0 vs arg1

Reply:
No, it's good to have the third test - we want to make sure that, with that flag on, we have the same schema for the result in all cases.

Reply:
Then we have a Bug here too! Coz, Round function returns return type same as arg0 if arg0 is table.

Reply:
I have filed issue https://github.com/microsoft/Power-Fx/issues/709 for this bug.